File: 2018/day01/part1.clj
(require '[clojure.string :as str])

(defn parse-int [v] (Integer/parseInt v))

(defn get-input []
  (map parse-int (str/split (slurp "input.txt") #"\n")))

(defn part1 [input] (reduce + input))

(defn part2 [input]
  (loop [freqs #{}
         current 0
         tail (cycle input)]
    (if (contains? freqs current)
      current
      (recur
        (conj freqs current)
        (+ current (first tail))
        (rest tail)))))

(let [input (get-input)]
  (println "Part 1:" (part1 input))
  (println "Part 2:" (part2 input)))

File: 2018/day01/part1.py
from functools import reduce


def part1(offsets):
    return reduce(lambda x, y: x + y, offsets)


def part2(offsets):
    seen = set()
    current = 0
    while True:
        for offset in offsets:
            current += offset
            if current in seen:
                return current
            seen.add(current)


with open('input.txt','r') as f:
    offsets = list(map(lambda x: int(x), f))

    print("Part 1:", part1(offsets))
    print("Part 2:", part2(offsets))
